cljfmt/src/cljfmt/core.cljc

```diff
@@ -674,17 +680,24 @@
 (defn align-map-columns [form]
   (transform form edit-all z/map? align-columns))
 
-(defn- aligned-form? [zloc aligned-forms context]
-  (and (z/list? (z/up zloc))
-       (some (fn [[k indexes]]
-               (and (form-matches-key? zloc k context)
-                    (contains? (set indexes) (dec (index-of zloc)))))
-             aligned-forms)))
+(defn- matching-form? [zloc aligned-forms context]
+  (when-let [f (cond
+                 (z/list? zloc)
+                 (fn [[k indexes]]
+                   (and (form-matches-key? (z/down zloc) k context)
+                        (= :all indexes)))
+
+                 (z/list? (z/up zloc))
+                 (fn [[k indexes]]
+                   (and (set? indexes)
+                        (form-matches-key? zloc k context)
+                        (contains? (set indexes) (dec (index-of zloc))))))]
+    (some f aligned-forms)))
```

Comment:
I think we can make this a touch clearer and more efficient by splitting this into two functions:

```clojure
(defn- matching-form-index? [zloc [k indexes] context]
  (if (= :all indexes)
    (and (z/list? zloc)
         (form-matches-key? (z/down zloc) k context))
    (and (z/list? (z/up zloc))
         (form-matches-key? zloc k context)
         (contains? (set indexes) (dec (index-of zloc))))))

(defn- matching-form? [zloc form-indexes context]
  (and (or (z/list? zloc) (z/list (z/up zloc)))
       (some #(matching-form-index? zloc % context) form-indexes)))
```

So we have a quick check first, just to see if we're in a list or a direct child of a list, then we iterate through the form indexes to find a match.